src/reanimated2/SensorContainer.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  SensorType,
+  SensorConfig,
+  Value3D,
+  ValueRotation,
+  ShareableRef,
+  AnimatedSensor,
+} from './commonTypes';
+import Sensor from './Sensor';
+
+export class SensorContainer {
+  private nativeSensors: Map<number, Sensor<any>> = new Map();
+
+  getSensorId(sensorType: SensorType, config: SensorConfig) {
+    return (
+      sensorType * 100 +
+      config.iosReferenceFrame * 10 +
+      Number(config.adjustToInterfaceOrientation)
+    );
+  }
+
+  registerSensor<T>(
+    sensorType: SensorType,
+    sensorRef: React.MutableRefObject<AnimatedSensor>,
+    handler: ShareableRef<T> | ((data: Value3D | ValueRotation) => void)
+  ): number {
+    const config = sensorRef.current.config;
+    const sensorId = this.getSensorId(sensorType, config);
+    console.log(this.nativeSensors.size);
+
+    if (!this.nativeSensors.has(sensorId)) {
+      const newSensor = new Sensor(
+        sensorType,
+        config,
+        sensorRef.current.sensor,
+        handler
+      );
+      this.nativeSensors.set(sensorId, newSensor);
+    }
+
+    const sensor = this.nativeSensors?.get(sensorId);
+
+    if (!sensor?.isRunning()) {
+      if (!sensor?.initialize()) {
+        return -1;
+      }
+    }
+    sensorRef.current.sensor = sensor?.getSharedValue();
```

Comment:
This part is a bit confusing for me 🤔 
1. You create the new shared value in `useAnimatedSensor` for every `useAnimatedSensor` call.
```
export function useAnimatedSensor(
  sensorType: SensorType,
  userConfig?: Partial<SensorConfig>
): AnimatedSensor {
  const ref = useRef<AnimatedSensor>({
    sensor: initSensorData(sensorType),
```
2. Then we overriding it by previous registered sharedValue for this type of sensor.:
```
sensorRef.current.sensor = sensor?.getSharedValue();
```
Why we can't create a new shared value once in `Sensor` constructor?

Reply:
It's because we can have situation when we attach a native listener to shared value (e.g. `useAnimatedStyle`) before sensor is registered or we still have shared value from previous one -  this problem was mainly with fast refresh.